### ISSUE TITLE
inferPlanarMaxCut of python binding

### ISSUE THREAD

**Issue by tttamaki** [opengm/opengm#397]
When building the python binding with `WITH_BLOSSOM5=OFF` and `WITH_PLANARITY=OFF`, an error occurs in `submodel2.hxx` because `opengm::external::PlanarMaxCut` is not defined.

Here is a small fix.

```diff:test
diff --git a/include/opengm/inference/cgc/submodel2.hxx b/include/opengm/inference/cgc/submodel2.hxx
index a6833f3..e97068e 100644
--- a/include/opengm/inference/cgc/submodel2.hxx
+++ b/include/opengm/inference/cgc/submodel2.hxx
@@ -810,6 +810,7 @@ template<class GM>
 typename SubmodelCGC<GM>::ValueType SubmodelCGC<GM>::inferPlanarMaxCut(
 
 ){  
+#if defined(WITH_BLOSSOM5) && defined(WITH_PLANARITY)
     opengm::external::PlanarMaxCut solver(numSubVar_, numSubFactors_);
     for(size_t i=0; i<numSubFactors_; ++i){
         solver.addEdge(localFactorVis_(i,0),localFactorVis_(i,1), -1.0*localLambdas_[i]);
@@ -826,6 +827,10 @@ typename SubmodelCGC<GM>::ValueType SubmodelCGC<GM>::inferPlanarMaxCut(
             value+=localLambdas_[i];
     }
     return value;
+#else
+   throw opengm::RuntimeError("inferPlanarMaxCut needs WITH_BLOSSOM5 and WITH_PLANARITY");
+   return 0.0;
+#endif
 }
 
```

**Comment by joergkappes** [comment]
Hi,
thanks for pointing this out. 
I am still waiting for a PR of Thorsten, which maybe fix this bug, too.
If not I will add your bugfix before lunching the next version end of the week.

Best Joerg

**Comment by DerThorsten** [comment]
@joergkappes  you can merge that, i need a tiny bit to make my branch up to date!
And my PR is somehow not related

**Comment by joergkappes** [comment]
fixed with #50e7ef63c36f768c773ed432bcbd126ca0f49c71
